core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/booleans/BooleanPrimitiveValueSemanticsProvider.java

```diff
@@ -56,21 +56,21 @@ public ObjectAdapter getDefault(final ObjectAdapter inObject) {
     // BooleanValueFacet impl
     // //////////////////////////////////////////////////////////////////
 
-    @Override
-    public void reset(final ObjectAdapter object) {
-        object.replacePojo(Boolean.FALSE);
-    }
-
-    @Override
-    public void set(final ObjectAdapter object) {
-        object.replacePojo(Boolean.TRUE);
-    }
-
-    @Override
-    public void toggle(final ObjectAdapter object) {
-        final boolean current = ((Boolean) object.getObject()).booleanValue();
-        final boolean toggled = !current;
-        object.replacePojo(Boolean.valueOf(toggled));
-    }
+//    @Override
+//    public void reset(final ObjectAdapter object) {
+//        object.replacePojo(Boolean.FALSE);
+//    }
+//
+//    @Override
+//    public void set(final ObjectAdapter object) {
+//        object.replacePojo(Boolean.TRUE);
+//    }
+//
+//    @Override
+//    public void toggle(final ObjectAdapter object) {
+//        final boolean current = ((Boolean) object.getObject()).booleanValue();
+//        final boolean toggled = !current;
+//        object.replacePojo(Boolean.valueOf(toggled));
+//    }
```

Comment:
yep.  I suspect that quite a bit of ValueSemanticsProvider is redundant.  It was implemented before either of the two current viewers (Wicket and RO), and to be honest they both have their own way of doing some of the responsiblities of VSPs, the parsing/encoding of text into value... in the case of Wicket, using its IConverter<> API, for RO I can't remember.

So, instead of VSPs, we should maybe say its the responsibility of the viewer to obtain values from its UI widgets, and just have the metamodel deal in terms of the actual objects (Strings, booleans, dates etc).